Rewrite SVG slope page (#34767)
mdn/content: files/en-us/web/svg/attribute/slope/index.md

title: slope
slug: Web/SVG/Attribute/slope
page-type: svg-attribute
browser-compat: svg.elements.feFuncR
---

{{SVGRef}}

> The **`slope`** attribute defines the values for linear filters, such as brightness. The `slope` attribute is suppported by children of the {{SVGElement("feComponentTransfer")}} filter primitive, including the `<feFuncR>`, `<feFuncB>`, `<feFuncG>`, and `<feFuncA>` transfer function. When the `type` of transfer function is `linear`, the `slope` defines the slope of the linear function.

Reviewer: ```suggestion
The **`slope`** attribute defines the values for linear filters, such as brightness. The `slope` attribute is supported by children of the {{SVGElement("feComponentTransfer")}} filter primitive, including the `<feFuncR>`, `<feFuncB>`, `<feFuncG>`, and `<feFuncA>` transfer functions. When the `type` of a transfer function is `linear`, the `slope` defines the slope of the linear function.
```

> You can use this attribute with the following SVG filter primitive transfer function elements when `type="linear"` is set:

Reviewer: ```suggestion
You can use this attribute with the following SVG filter primitive component transfer function elements when `type="linear"` is set:
```

- {{SVGElement("feFuncR")}}

> - {{SVGElement("feFuncB")}}

Reviewer: ```suggestion
- {{SVGElement("feFuncG")}}
```

title: slope
slug: Web/SVG/Attribute/slope
page-type: svg-attribute
browser-compat: svg.elements.feFuncR
spec-urls: https://drafts.fxtf.org/filter-effects/#element-attrdef-fecomponenttransfer-slope
---

{{SVGRef}}

> The **`slope`** attribute defines the values for linear filters, such as brightness. The `slope` attribute is supported by children of the {{SVGElement("feComponentTransfer")}} filter primitive, including the `<feFuncR>`, `<feFuncB>`, `<feFuncG>`, and `<feFuncA>` transfer functions. When the `type` of a transfer function is `linear`, the `slope` defines the slope of the linear function.

Reviewer 2: ```suggestion
The **`slope`** attribute defines the values for linear filters, such as brightness.

The `slope` attribute is supported by children of the {{SVGElement("feComponentTransfer")}} filter primitive, including the `<feFuncR>`, `<feFuncB>`, `<feFuncG>`, and `<feFuncA>` transfer functions. When the `type` of a transfer function is `linear`, the `slope` defines the slope of the linear function.
```

title: slope
slug: Web/SVG/Attribute/slope
page-type: svg-attribute
browser-compat: svg.elements.feFuncR
spec-urls: https://drafts.fxtf.org/filter-effects/#element-attrdef-fecomponenttransfer-slope
---

{{SVGRef}}

The **`slope`** attribute defines the values for linear filters, such as brightness.

> The `slope` attribute is supported by children of the {{SVGElement("feComponentTransfer")}} filter primitive, including the `<feFuncR>`, `<feFuncB>`, `<feFuncG>`, and `<feFuncA>` transfer functions. When the `type` of a transfer function is `linear`, the `slope` defines the slope of the linear function.

Author: ```suggestion
The `slope` attribute is supported by children of the {{SVGElement("feComponentTransfer")}} filter primitive, including the `feFunc-RGBA` transfer functions. When the `type` of a transfer function is `linear`, the `slope` defines the slope of the linear function.
```

You can use this attribute with the following SVG component transfer function elements when `type="linear"` is set:

- {{SVGElement("feFuncR")}}
- {{SVGElement("feFuncG")}}
- {{SVGElement("feFuncB")}}

> - {{SVGElement("feFuncA")}}

Reviewer 3: Same here

    </tr>
    <tr>
      <th scope="row">Default value</th>
      <td><code>1</code></td>
    </tr>
    <tr>
      <th scope="row">Animatable</th>
      <td>Yes</td>
    </tr>
  </tbody>
</table>

- `<number>`

>   - : This value indicates the linear slope of the transfer function.

Reviewer: ```suggestion
  - : This value indicates the slope of the linear transfer functions.
```

## Examples

In this example, a gradient box has two text elements with linear filters applied. The slopes of the filter transfer functions of each filter differ.

```html
<svg
  width="8cm"
  height="4cm"
  viewBox="0 0 800 400"
  xmlns="http://www.w3.org/2000/svg"
  version="1.1">
  <title>Examples of feComponentTransfer operations</title>
  <desc>
    Text strings showing the effects of the slope attribute of the
    feComponentTransfer filter function linear options.
  </desc>
  <defs>
    <linearGradient
      id="MyGradient"
      gradientUnits="userSpaceOnUse"
      x1="100"
      y1="0"
      x2="600"
      y2="0">
      <stop offset="0" stop-color="#ff0000" />
      <stop offset="1" stop-color="#0000ff" />
    </linearGradient>
    <filter
      id="Linear1"
      filterUnits="objectBoundingBox"
      x="0%"
      y="0%"
      width="100%"
      height="100%">
      <feComponentTransfer>

>         <feFuncR type="linear" slope="2" />
>         <feFuncG type="linear" slope="25" />
>         <feFuncB type="linear" slope="5" />

Reviewer: ```suggestion
        <feFuncR type="linear" slope="1" />
        <feFuncG type="linear" slope="1" />
        <feFuncB type="linear" slope="1" />
```
For better understanding we can have a comparison with the default value `1`

      </feComponentTransfer>
    </filter>
    <filter
      id="Linear2"
      filterUnits="objectBoundingBox"
      x="0%"
      y="0%"
      width="100%"
      height="100%">
      <feComponentTransfer>

>         <feFuncR type="linear" slope="1" />
>         <feFuncG type="linear" slope="0.5" />
>         <feFuncB type="linear" slope="1" />

Reviewer: ```suggestion
        <feFuncR type="linear" slope="2" />
        <feFuncG type="linear" slope="2.5" />
        <feFuncB type="linear" slope="2" />
```
related to previous comment.

    </tr>
    <tr>
      <th scope="row">Default value</th>
      <td><code>1</code></td>
    </tr>
    <tr>
      <th scope="row">Animatable</th>
      <td>Yes</td>
    </tr>
  </tbody>
</table>

- `<number>`

>   - : This value indicates the slope of the linear component transfer functions.

Reviewer 2: Is this constrained to any range? From this I would assume it is a range between =/- infinite.

I understand the slope changes the the gradient, but it is very abstract from this definition, and the example just shows me it occurs, not how I can "understand it".

Author: the spec just says `<number>`, and the two books i consulted didn't provide any additional information

Reviewer 2: OK, That's enough for now then.
